File: ApplicationServer/ApplicationServer/src/main/java/licenta/applicationserver/entities/SensorData.java
package licenta.applicationserver.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "sensor_data")
public class SensorData {
    //describes the entity that represents the sensor data

    //Primary Key
    @Id
    private Integer environmentId;

    @OneToOne
    @MapsId
    @JoinColumn(name = "environment_id", nullable = false)
    private Environment environment;

    @Column(name = "temperature")
    private Double temperature;

    @Column(name = "humidity")
    private Double humidity;

    @Column(name = "luminosity")
    private Double luminosity;

    @Column(name = "pressure")
    private Double pressure;

    @Column(name = "soil_moisture")
    private Double soilMoisture;
}

File: AgricultureAutomationApp/app/src/main/java/com/example/agricultureautomationapp/models/SensorData.kt
package com.example.agricultureautomationapp.models

class SensorData {
    var temperature: Double
    var humidity: Double
    var luminosity: Double
    var pressure: Double
    var soilMoisture: Double

    constructor(temperature: Double, humidity: Double, luminosity: Double, pressure: Double, soilMoisture: Double) {
        this.temperature = temperature
        this.humidity = humidity
        this.luminosity = luminosity
        this.pressure = pressure
        this.soilMoisture = soilMoisture
    }
}